src/pages/home/ReportScreen.js

```diff
@@ -106,6 +111,9 @@ class ReportScreen extends React.Component {
     }
 
     componentDidUpdate(prevProps) {
+        if (this.state.isGhostScreenVisible && this.props.isLoadingReportData === false) {
```

Comment:
**question:** Why are we hiding the skeleton screen only after report data is loaded?

**question:** Can you also explain which "report data" is loading in this case?

Reply:
This is to determine that the API has loaded the data and stored it into Onyx. I had checked about it here, https://github.com/Expensify/App/pull/7746#issuecomment-1050032885, and tried to check the usages from `ReportActionsView.isLoadingReportData`

Reply:
Hmm sorry I am not sure if this makes sense. We might have the report actions for the report yet we are waiting for all of the report data to load? Why?

Reply:
Okay, let me take a step back and recheck this logic.  For the skeleton UI to show/hide, I am adding isSkeletonViewVisible. To set/unset I need to be able to identify if the chat data is being loaded.

Need help with the following:
1. At the moment, I am using `IS_LOADING_REPORT_DATA`.  Is this the correct flag? If not then what should be the right one.
2. One issue with  `IS_LOADING_REPORT_DATA`, is that it gets set for every time the API is called, hence I am using the logic here, https://github.com/Expensify/App/pull/8042#discussion_r863209047 to hide it and ensure it doesn't show up again
3. Existing `isLoading` existed beforehand where `prepareTransition` internally calls `setTimeout` to render the loader smoothly. What are your thoughts of retaining this logic?

Reply:
1. `IS_LOADING_REPORT_DATA` is set when while we are loading the big list of reports (search for the flag and we can see where it happens exactly).
2. Yes, this seems bad because if we are "reconnecting" and refetching the reports then a chat will show as "loading"? That sounds incorrect. Why should the report show as loading when we are loading all report summaries?
3. I'm not sure what you are suggesting here. I have no thoughts about retaining this logic because I don't fully understand it's purpose. Are you able to investigate and produce an explanation and then propose a way forward?

Reply:
Okay, so I checked the logic again. We have a function called `Report.fetchActions`, which pulls the data for the given `reportId`. This gets called on load for `ReportActionsView`. 

https://github.com/Expensify/App/blob/724ccb46ca4b2f48f6052592cbacc7ffe9528c5c/src/libs/actions/Report.js#L951-L956

I am proposing to add a flag here which will help figure out if the data is being loaded when the offset is undefined.
```
function fetchActions(reportID, offset) {
    const reportActionsOffset = !_.isUndefined(offset) ? offset : -1;
    if(reportActionsOffset === -1) {
       Onyx.set(`${ONYXKEYS.IS_SKELETON_VIEW_VISIBLE}`, true);
    }
    
    ...
    ...
       Onyx.set(`${ONYXKEYS.IS_SKELETON_VIEW_VISIBLE}`, false); // Add this
       Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, indexedData); // existing code line 972
```

then remove those additional loading checks and state variables, and replace them with the above Onyx key.

----

I thought of using the existing `IS_LOADING_REPORT_ACTIONS`, but it conflicts with load more, and we would have to unnecessarily add checks for those into state the way we are doing.

----
> Are you able to investigate and produce an explanation and then propose a way forward?

For point number 3, it seems the loader was added as a dummy only to show a small transition  (may be to look smoother). I feel we should get rid of it completely.

Here's the attached video **without** the `isLoading` effect.
https://user-images.githubusercontent.com/3069065/167313000-f7714b45-82db-435d-ac34-aa949d6a6510.mp4

Here's the one **with** the current `isLoading` effect.
https://user-images.githubusercontent.com/3069065/167313016-5c4c46e9-603d-436f-9ad6-13b81d687cfd.mov

cc -  @Luke9389 @marcaaron @roryabraham @parasharrajat

Reply:
Hi Team,
Any feedback on the above approach?

Reply:
I don't have much idea about this. But I feel that it would be unnecessary when the actions are already synced in onyx and this is called for a safety check.

Reply:
>  it would be unnecessary when the actions are already synced i

As I said, I could definitely check it, but we use this for scrolling chats. So I thought if we store a separate flag for this, it would be better.

Reply:
>I am proposing to add a flag here which will help figure out if the data is being loaded when the offset is undefined.

Why do we need to "figure out if the data is being loaded when the offset is undefined"?

>For point number 3, it seems the loader was added as a dummy only to show a small transition (may be to look smoother). I feel we should get rid of it completely.

I would be in favor of this if we are replacing the loader with the skeleton UI.

Reply:
> Why do we need to "figure out if the data is being loaded when the offset is undefined"?

The first load of the data on component mount is via  `Report.fetchActions(this.props.reportID);` which calls the following:

https://github.com/Expensify/App/blob/a499e431215d0bf3fdfd62b28c8f3d92c977ec6c/src/libs/actions/Report.js#L951-L954

We're setting `IS_LOADING_REPORT_ACTIONS` in this function. 

Hence I said,
1. I could add a flag in `Onyx`, which will purely be used to show visibility of skeletal UI

3. or change the function call from `Report.fetchActions(this.props.reportID);` to `Report.fetchActionsWithLoadingState(reportId, offset)` and use `IS_LOADING_REPORT_ACTIONS`, and use the existing `isSkeletalViewVisible` state variable.

Let me know your thoughts.

> I would be in favor of this if we are replacing the loader with the skeleton UI.

Yeah, I also feel the same.

Reply:
I'm totally lost on what you are proposing here and why we should do it. Let's bring this conversation back to the issue since I don't think the original design is correct and it requires more feedback.

Reply:
Stale. We've updated the approach.